src/System.Diagnostics.TraceSource/tests/TraceSourceClassTests.cs

```diff
@@ -116,6 +116,44 @@ public void SwitchLevelTest(SourceLevels sourceLevel, TraceEventType messageLeve
             trace.TraceEvent(messageLevel, 0);
             Assert.Equal(expected, listener.GetCallCount(Method.TraceEvent));
         }
+
+        [Fact]
+        public void NullSourceName()
+        {
+            Assert.Throws<ArgumentNullException>(() => new TraceSource(null));
+        }
+
+        [Theory]
+        [InlineData(SourceLevels.Off)]
+        [InlineData(SourceLevels.Critical)]
+        [InlineData(SourceLevels.Error)]
+        [InlineData(SourceLevels.Warning)]
+        [InlineData(SourceLevels.Information)]
+        [InlineData(SourceLevels.Verbose)]
+        [InlineData(SourceLevels.All)]
+        public void NullSourceName(SourceLevels level)
+        {
+            Assert.Throws<ArgumentNullException>(() => new TraceSource(null, level));
+        }
+
+        [Fact]
+        public void EmptySourceName()
+        {
+            Assert.Throws<ArgumentException>(() => new TraceSource(string.Empty));
+        }
+
+        [Theory]
+        [InlineData(SourceLevels.Off)]
+        [InlineData(SourceLevels.Critical)]
+        [InlineData(SourceLevels.Error)]
+        [InlineData(SourceLevels.Warning)]
+        [InlineData(SourceLevels.Information)]
+        [InlineData(SourceLevels.Verbose)]
+        [InlineData(SourceLevels.All)]
```

Comment:
All of these cases aren't necessary; there's nothing about the SourceLevels that impacts the argument validation of the first argument.  You can just choose one source level to use and add the assert to the previous Fact.